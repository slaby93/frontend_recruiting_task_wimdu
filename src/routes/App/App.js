import React from 'react'
import { connect } from 'react-redux';
import Button from './../../components/Button'

export const App = () => {
  return (<div>
    <Button>Fetch users</Button>
  </div>);
}

export default App