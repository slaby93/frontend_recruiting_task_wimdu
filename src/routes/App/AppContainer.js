import React from 'react';
import App from './App'
import { connect } from 'react-redux'
import { FETCH_USERS_REQUEST } from './../../reducers/users'

const AppContainer = (props) => {
  return <App {...props} />
}

const mapStateToProps = ({ users }) => {
  const isRequesting = users.isRequesting;

  if (isRequesting) {
    return { isRequesting }
  }

  return users
}

const mapDispatchToProps = (dispatch) => {
  return {
    fetchUsers: () => {
      dispatch(FETCH_USERS_REQUEST());
    }
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AppContainer);