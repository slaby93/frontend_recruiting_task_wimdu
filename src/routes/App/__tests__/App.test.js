import React from 'react';
import { App } from './../App'
import { shallow, mount } from 'enzyme'

describe('<App/>', () => {
  test('basic render', () => {
    expect(shallow(<App />)).toMatchSnapshot();
  })
});