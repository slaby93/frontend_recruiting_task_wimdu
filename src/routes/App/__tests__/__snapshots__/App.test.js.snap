// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App/> basic render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <Button>
        Fetch users
      </Button>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "function",
      "props": Object {
        "children": "Fetch users",
      },
      "ref": null,
      "rendered": "Fetch users",
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <Button>
          Fetch users
        </Button>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "function",
        "props": Object {
          "children": "Fetch users",
        },
        "ref": null,
        "rendered": "Fetch users",
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
